capstone/src/test/java/com/google/sps/servlets/CreateNewUserServletTest.java

```diff
@@ -0,0 +1,124 @@
+            new LocalDatastoreServiceTestConfig()
+                .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+            new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  private static ArrayList<String> INTERESTS_LIST = new ArrayList<String>( 
+      Arrays.asList("Testing", "Dancing"));
+  private static final User USER_1 = new User(USER_ID, "Test", "McTest", USER_EMAIL, 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ INTERESTS_LIST);
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private CreateNewUserServlet createNewUserServlet;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    createNewUserServlet = new CreateNewUserServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_addNewUser() throws Exception {
+    when(mockRequest.getParameter("first")).thenReturn(USER_1.getFirstName());
+    when(mockRequest.getParameter("last")).thenReturn(USER_1.getLastName());
+    when(mockRequest.getParameter("phone")).thenReturn(USER_1.getPhoneNumber());
+    when(mockRequest.getParameter("interests")).thenReturn("Testing, Dancing");
+
+    createNewUserServlet.doPost(mockRequest, mockResponse);
+    Key userKey = KeyFactory.createKey("User", USER_ID);
+    Entity user = datastore.get(userKey);
+    
+    assertTrue(USER_1.equals(User.fromEntity(user)));
+  }
+
+  @Test(expected = EntityNotFoundException.class)
+  public void doPost_userNotLoggedIn() throws Exception {
+    helper.setEnvIsLoggedIn(false);
+
+    when(mockRequest.getParameter("first")).thenReturn(USER_1.getFirstName());
+    when(mockRequest.getParameter("last")).thenReturn(USER_1.getLastName());
+    when(mockRequest.getParameter("phone")).thenReturn(USER_1.getPhoneNumber());
+    when(mockRequest.getParameter("interests")).thenReturn("Testing, Dancing");
+
+    createNewUserServlet.doPost(mockRequest, mockResponse);
+
+    Key userKey = KeyFactory.createKey("User", USER_ID);
+    datastore.get(userKey); 
+
+    helper.setEnvIsLoggedIn(true);
```

Comment:
There don't appear to be any assertions in this test. If this is intended, I would add a comment after the line of code that should trigger the exception, and ideally the test would end at that comment, since you should try to just test the exception, any other testing should be done in another test.

Reply:
Got it, will do